cmd/explaintest/r/access_path_selection.result

```diff
@@ -0,0 +1,91 @@
+  │ └─IndexScan_16	3333.33	cop	table:pay_daily, index:update_source, a_id, range:(12565,+inf], keep order:false, stats:pseudo
+  └─TableScan_17	1107.78	cop	table:pay_daily, keep order:false
+explain select id from pay_daily where date_key>12565;
+id	count	task	operator info
+Projection_4	3333.33	root	test.pay_daily.id
+└─IndexLookUp_10	3333.33	root	
+  ├─IndexScan_8	3333.33	cop	table:pay_daily, index:date_key, range:(12565,+inf], keep order:false, stats:pseudo
+  └─TableScan_9	3333.33	cop	table:pay_daily, keep order:false, stats:pseudo
+explain select id from pay_daily where date_key>12565 and io_type=1;
+id	count	task	operator info
+Projection_4	3.33	root	test.pay_daily.id
+└─IndexLookUp_11	3.33	root	
+  ├─IndexScan_8	3333.33	cop	table:pay_daily, index:date_key, range:(12565,+inf], keep order:false, stats:pseudo
+  └─Selection_10	3.33	cop	eq(test.pay_daily.io_type, 1)
+    └─TableScan_9	3333.33	cop	table:pay_daily, keep order:false, stats:pseudo
+explain select id from pay_daily where date_key>12565 and a_id<159298363;
+id	count	task	operator info
+Projection_4	1107.78	root	test.pay_daily.id
+└─IndexLookUp_19	1107.78	root	
+  ├─Selection_18	1107.78	cop	gt(test.pay_daily.date_key, 12565)
+  │ └─IndexScan_16	3323.33	cop	table:pay_daily, index:a_id, b_id, date_key, range:[-inf,159298363), keep order:false, stats:pseudo
+  └─TableScan_17	1107.78	cop	table:pay_daily, keep order:false
+CREATE TABLE `access_path_selection` (
+`a` int,
+`b` int,
+KEY `IDX_a` (`a`),
+KEY `IDX_b` (`b`),
+KEY `IDX_ab` (`a`, `b`)
+);
+explain select a, b from access_path_selection where a < 3;
+id	count	task	operator info
+IndexReader_12	3323.33	root	index:IndexScan_11
+└─IndexScan_11	3323.33	cop	table:access_path_selection, index:a, b, range:[-inf,3), keep order:false, stats:pseudo
+explain select a, b from access_path_selection where b < 3;
+id	count	task	operator info
+IndexLookUp_10	3323.33	root	
+├─IndexScan_8	3323.33	cop	table:access_path_selection, index:b, range:[-inf,3), keep order:false, stats:pseudo
+└─TableScan_9	3323.33	cop	table:access_path_selection, keep order:false, stats:pseudo
+explain select a, b from access_path_selection where a < 3 and b < 3;
+id	count	task	operator info
+IndexReader_18	1104.45	root	index:Selection_17
+└─Selection_17	1104.45	cop	lt(test.access_path_selection.b, 3)
+  └─IndexScan_16	3323.33	cop	table:access_path_selection, index:a, b, range:[-inf,3), keep order:false, stats:pseudo
+CREATE TABLE `wrong_index_selection` (
+`a` int,
+`b` int,
+`c` int,
+INDEX idx_a(a),
+INDEX idx_ab(a,b)
+);
+insert into wrong_index_selection values (1, 1, 1);
+insert into wrong_index_selection values (2, 2, 2);
+insert into wrong_index_selection values (3, 3, 3);
+analyze table wrong_index_selection index idx_ab;
+explain select * from wrong_index_selection where a=1 and b=1 and c=1;
+id	count	task	operator info
+IndexLookUp_11	0.80	root	
+├─IndexScan_8	1.00	cop	table:wrong_index_selection, index:a, range:[1,1], keep order:false
```

Comment:
I think in this case, the estimation row count is right, and it is not a wrong choice.

Reply:
The case has been updated, how about now?